src/django/api/serializers/v1/moderation_event_update_serializer.py

```diff
@@ -78,12 +84,61 @@ def __validate_status(self, value):
             })
         return value
 
+    def validate(self, data):
+        status = data.get('status')
+
+        if status != ModerationEvent.Status.REJECTED:
+            return data
+
+        cleaned = data.get('action_reason_text_cleaned')
+        raw = data.get('action_reason_text_raw')
+        errors = []
+
+        if not cleaned:
+            errors.append({
+                "field": "action_reason_text_cleaned",
+                "detail": (
+                    "This field is required when rejecting a moderation "
+                    "event."
+                )
+            })
+        elif len(cleaned) < self.MIN_REASON_TEXT_LENGTH:
+            errors.append({
+                "field": "action_reason_text_cleaned",
+                "detail": "This field must be at least 30 characters."
+            })
+
+        if not raw:
+            errors.append({
+                "field": "action_reason_text_raw",
+                "detail": (
+                    "This field is required when rejecting a moderation "
+                    "event."
+                )
+            })
+        elif len(raw) < self.MIN_REASON_TEXT_LENGTH:
+            errors.append({
+                "field": "action_reason_text_raw",
+                "detail": "This field must be at least 30 characters."
+            })
```

Comment:
This is just a comment for the future, no need to change it in this PR. I would suggest assigning the variable right before using it instead of at the top of the file, that means assigning the `raw` value right before this if else statement, like this:

```python
raw = data.get('action_reason_text_raw')

if not raw:
            errors.append({
                "field": "action_reason_text_raw",
                "detail": (
                    "This field is required when rejecting a moderation "
                    "event."
                )
            })
        elif len(raw) < self.MIN_REASON_TEXT_LENGTH:
            errors.append({
                "field": "action_reason_text_raw",
                "detail": "This field must be at least 30 characters."
            })
```